### PR TITLE
Abstract service and Service util

### PR DESCRIPTION
**Pull request by jdahmubed**

### CODE REVIEW

File: src/service/AbstractService.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
+package service;
+
+import config.ServiceConfig;
+import interfaces.State;
+import enums.ClientAuthenticationMethod;
+import enums.EndpointName;
+import enums.HttpMethod;
+import enums.SerializationType;
+import enums.ServiceName;
+import helper_classes.HttpArguments;
```
Comment (manu-sinha) [inline src/service/AbstractService.java]: Please follow Java style guide .. use helperclasses
refer - http://go/java-style#s5.2.1-package-names

File: src/service/AbstractService.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+import helper_classes.HttpHeader;
+import java.util.Map;
+import service_context.ServiceContext;
```
Comment (manu-sinha) [inline src/service/AbstractService.java]: Please follow Java style guide .. use servicecontext
refer - http://go/java-style#s5.2.1-package-names

**Review by manu-sinha — CHANGES_REQUESTED** [review]

File: src/utils/ServiceUtil.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ * This class has utility methods for various services
+ **/
+public class ServiceUtil {
+    /**
+     Pick out the fragment or query part from a URL.
+     * @param url a URL possibly containing a query or a fragment part
+     * @return the query/fragment part
+     **/
+    public static String getUrlInfo(String url) {
+
```
Comment (manu-sinha) [inline src/utils/ServiceUtil.java]: When you create a PR, do you make sure all the code is compiling? Looking at this class I don't think so.
For this first PR, I will review only if the code is compiling
From second PR onwards, I will review only if code is compiling and there are unit tests.
@lccodes FYI

**Review by manu-sinha — CHANGES_REQUESTED** [review]
Please confirm that all the code in this PR are compiling.

File: src/service/AbstractService.java (hunk 1 above)
Comment (jdahmubed) [reply src/service/AbstractService.java]: done

File: src/service/AbstractService.java (hunk 2 above)
Comment (jdahmubed) [reply src/service/AbstractService.java]: done

File: src/utils/ServiceUtil.java (hunk 3 above)
Comment (jdahmubed) [reply src/utils/ServiceUtil.java]: add return null to all

File: src/org/oidc/service/AbstractService.java [hunk 4]
```diff
@@ -0,0 +1,215 @@
+package org.oidc.service;
+
+import org.oidc.service.base.ServiceConfig;
+import org.oidc.service.data.State;
+import org.oidc.service.base.enums.ClientAuthenticationMethod;
```
Comment (manu-sinha) [inline src/org/oidc/service/AbstractService.java]: please update the imports

**Review by manu-sinha — CHANGES_REQUESTED** [review]

File: src/org/oidc/service/AbstractService.java (hunk 4 above)
Comment (jdahmubed) [reply src/org/oidc/service/AbstractService.java]: done

File: src/org/oidc/service/AbstractService.java [hunk 5]
```diff
@@ -0,0 +1,215 @@
+package org.oidc.service;
+
+import org.oidc.service.base.ServiceConfig;
```
Comment (manu-sinha) [inline src/org/oidc/service/AbstractService.java]: Could you please order the imports? Follow https://google.github.io/styleguide/javaguide.html#s3.3.3-import-ordering-and-spacing
@lccodes FYI

**Review by manu-sinha — CHANGES_REQUESTED** [review]

File: src/org/oidc/service/AbstractService.java (hunk 5 above)
Comment (jdahmubed) [reply src/org/oidc/service/AbstractService.java]: reorganized imports

**Review by manu-sinha — APPROVED** [review]

**Pull request merged**